test/CommandLineUtils.Tests/MappedOptionAttributeTests.cs

```diff
@@ -0,0 +1,329 @@
+
+            [Option] public int NoEdit { get; }
+        }
+
+        [Fact]
+        public void ThrowsWhenLongOptionNamesAreAmbiguous()
+        {
+            var ex = Assert.Throws<InvalidOperationException>(
+                () => Create<AmbiguousLongOptionName>());
+
+            Assert.Equal(
+                Strings.OptionNameIsAmbiguous("no-edit",
+                    typeof(AmbiguousLongOptionName).GetProperty("NoEdit"),
+                    typeof(AmbiguousLongOptionName).GetProperty("ManuallySetToNoEdit")),
+                ex.Message);
+        }
+        */
+
+        private class BothOptionAndArgument
+        {
+            [MappedOption(nameof(NotPossible))]
+            [Argument(0)]
+            public int NotPossible { get; }
+        }
+
+        [Fact]
+        public void ThrowsWhenOptionAndArgumentAreSpecified()
+        {
+            FluentActions.Invoking(() => Create<BothOptionAndArgument>())
+                .Should().Throw<InvalidOperationException>()
+                .And
+                .Message.Should().Be(Strings.BothOptionAndArgumentAttributesCannotBeSpecified(
+                    typeof(BothOptionAndArgument).GetProperty("NotPossible")));
+        }
+
+
+        private class OptionHasDefaultValues
+        {
+            [MappedOption("-a1", "z")] public string Arg1 { get; } = "a";
+
+            [MappedOption("-a2", new[] { "y" })] public string[] Arg2 { get; } = new[] { "b", "c" };
+
+            [MappedOption("-a3", true)] public bool? Arg3 { get; }
+        }
+
+        [Fact]
+        public void KeepsDefaultValues()
+        {
+            using var assertionScope = new AssertionScope();
+
+            {
+                var app1 = Create<OptionHasDefaultValues>();
+                app1.Parse("-a1", "-a2");
+                app1.Model.Arg1.Should().Be("z");
+                app1.Model.Arg2.Should().BeEquivalentTo("y");
+                app1.Model.Arg3.Should().NotHaveValue();
+            }
+
+            {
+                var app2 = Create<OptionHasDefaultValues>();
+                app2.Parse("-a1");
+                app2.Model.Arg1.Should().Be("z");
+                app2.Model.Arg2.Should().BeEquivalentTo("b", "c");
+                app2.Model.Arg3.Should().NotHaveValue();
+            }
+
+            {
+                var app3 = Create<OptionHasDefaultValues>();
+                app3.Parse();
+                app3.Model.Arg1.Should().Be("a");
+                app3.Model.Arg2.Should().BeEquivalentTo("b", "c");
+                app3.Model.Arg3.Should().NotHaveValue();
+            }
+
+            {
+                var app4 = Create<OptionHasDefaultValues>();
+                app4.Parse("-a3");
+                app4.Model.Arg1.Should().Be("a");
+                app4.Model.Arg2.Should().BeEquivalentTo("b", "c");
+                app4.Model.Arg3.Should().BeTrue();
+            }
+        }
+
+        private class PrivateSetterProgram
```

Comment:
This, and the next 130 lines of code, appear to be copy/pasted from OptionAttributeTests. Can you delete instead of duplicating this content?